app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,50 @@
+class BookingsController < ApplicationController
+  def index
+    @bookings = Bookings.all
+  end
+
+  def new
+    @listing = Listing.find(params[:listing_id])
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @listing = Listing.find(params[:listing_id])
+    @booking.listing = @listing
+    if @booking.save
+      redirect_to booking_path(@booking), notice: 'Booking was successfully created!'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @booking = Booking.find(params[:id])
+  end
+
+  def edit
+    @booking = Booking.find(params[:id])
+  end
+
+  def update
+    @booking = Booking.find(params[:id])
+    if @booking.update(booking_params)
+      redirect_to booking_path(@booking), notice: 'Booking was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @booking = Booking.find(params[:id])
+    @booking.destroy
+    redirect_to bookings_path, notice: 'Booking was successfully deleted.'
```

Comment:
The path name here will be `listing_bookings_path`. Sorry we only had the routes.rb after you created this!

Reply:
No worries! Thanks for looking it over. I made the requested change.